lib/pool.js

```diff
@@ -27,21 +27,19 @@ function _addBackoff(factory) {
 
   factory.create = function __run() {
 
-    return originalCreateMethod().catch((e) =>
+    return originalCreateMethod().catch(function ___handleCreateError(e) {
 
-      new Bluebird(function ___backoffRunResolver(resolve, reject) {
-        const wait = backoff.next()
+      const wait = backoff.next()
 
-        debug(
-          '::_addBackoff - could not create resource - waiting %d ms - %s'
-        , wait
-        , e.message
-        )
+      debug(
+        '::_addBackoff - could not create resource - waiting %d ms - %s'
```

Comment:
I went back and forth on this.  I left it as the top level function because I figured it would be easier to find if you're looking for a top level function.